tetris.py

```diff
@@ -296,6 +319,35 @@ def get_block_shape(self):
 
         return coords[self.rotate_counter % len(coords)]
 
+    def get_upcoming_block_shape(self):
+        if self.upcoming_block_letter == 'I':
+            self.upcoming_block_center = (2, 0)
+        else:
+            self.upcoming_block_center = (1, 1)
+        (x, y) = self.upcoming_block_center
+
+        if self.upcoming_block_letter == "I":
+            coords = [(x - 2, y), (x - 1, y), (x, y), (x + 1, y)]
+
+        if self.upcoming_block_letter == "L":
+            coords = [(x - 1, y), (x, y), (x + 1, y), (x + 1, y - 1)]
+
+        if self.upcoming_block_letter == "L_rev":
+            coords = [(x - 1, y - 1), (x - 1, y), (x, y), (x + 1, y)]
+
+        if self.upcoming_block_letter == "O":
+            coords = [(x - 1, y), (x, y), (x, y - 1), (x - 1, y - 1)]
+
+        if self.upcoming_block_letter == "E":
+            coords = [(x - 1, y), (x, y), (x + 1, y), (x, y - 1)]
+
+        if self.upcoming_block_letter == "Z":
+            coords = [(x - 1, y - 1), (x, y - 1), (x, y), (x + 1, y)]
```

Comment:
There's no need to create a `coords` variable. Just return the lists directly. For example:

```suggestion
        if self.upcoming_block_letter == "Z":
            return [(x - 1, y - 1), (x, y - 1), (x, y), (x + 1, y)]
```

Reply:
This comment still applies, even after the copy/pasta cleanup.